### DISCUSSION TITLE
Explicit many-to-many relations with prisma

### DISCUSSION THREAD

**Discussion by axhamre** [hayes/pothos#407] [Q&A] [answered]
Is there any example how to use Pothos when having a many-to-many relation with Prisma, as prescribed with composite ID in their documentation? https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#explicit-many-to-many-relations

**Comment by hayes** [comment]
There isn't a great solution for this yet, but I am working on a new API that should solve this issue.

Would love any feedback you have on the proposed API described here: https://pothos-git-mh-complex-selections-mhayes.vercel.app/docs/plugins/prisma#indirect-relations-eg-join-tables.  I think this should work decently for this use case, but still getting some feedback to make sure this is a good solution.

**Reply by hayes** [reply]
This API is a little more complicated than the t.relation api, but is designed to support a wide variety of more advanced use cases.  It might be possible to create a simpler API for this specific use case, if this seems overly complex

**Reply by axhamre** [reply]
Looks great!

**Reply by hayes** [reply]
Forgot to update here, but this API mentioned above is now available.  It does require typescript 4.7 to work correctly

**Comment by hayes** [comment, marked as answer]
There is a new API that was recently added that should help resolve these more advanced patterns:  https://pothos-graphql.dev/docs/plugins/prisma#indirect-relations-eg-join-tables
